### ISSUE TITLE
Spellchecker is not activated by %spellcheck()% command 

### ISSUE THREAD

**Issue by goozak** [RealRaven2000/SmartTemplates#223]
Got this error message with a strange title (`__MSG_message_default_+title_`):

<img src="https://user-images.githubusercontent.com/454810/211840751-102b9dd3-89c8-4ccf-8447-175e6c05e659.png" width=75% height=75%>

Note that I'm only talking about the **dialog title** here, I'm pretty sure I know why I got this error — using the [LanguageTool](https://addons.thunderbird.net/thunderbird/addon/grammar-and-spell-checker/) add-on while having a `%spellcheck(en-CA,fr)%` command in my template (which worked before adding LanguageTool).

**Comment by goozak** [comment]
I've disabled all my other add-ons (including LanguageTool) and the issue is still there.  Not sure why I started having it at this time, although the last email I sent dates back to January 5th.

Here's the console content after a few second on a new message:
```
SmartTemplates [logTime init]
 SmartTemplate4.init() smartTemplate-util.js:688:13
SmartTemplates 13:57:23.528  [0 ms]	 
 SmartTemplate4.init() ends. smartTemplate-util.js:688:13
Uncaught TypeError: BondOpenPGP.isEnabled is not a function
    NotifyComposeBodyReadyST chrome://smarttemplate4/content/smartTemplate-main.js:363
    InitEditor chrome://messenger/content/messengercompose/MsgComposeCommands.js:10574
    observe chrome://messenger/content/messengercompose/MsgComposeCommands.js:4775
smartTemplate-main.js:363:85
SmartTemplates 13:57:23.829  [301 ms]	 
 NotifyComposeBodyReady smartTemplate-util.js:688:13
SmartTemplates 13:57:23.872  [43 ms]	 
 SmartTemplate4.notifyComposeBodyReady()
gMsgCompose type: 0 smartTemplate-util.js:688:13
SmartTemplates 13:57:23.875  [3 ms]	 
 spellChecker not available, retrying later...{ attempt 1 } smartTemplate-util.js:688:13
SmartTemplates 13:57:23.887  [12 ms]	 
 No deferred variables so we do not setup listeners... smartTemplate-util.js:688:13
SmartTemplates 13:57:23.910  [23 ms]	 
 SmartTemplate4.composer.load smartTemplate-util.js:688:13
SmartTemplates 13:57:23.911  [1 ms]	 
 composer.initTemplateMenu() ... smartTemplate-util.js:688:13
SmartTemplates 13:57:23.911  [0 ms]	 
 composer.initSnippetMenu() ... smartTemplate-util.js:688:13
SmartTemplates 13:57:25.878  [1967 ms]	 
 spellChecker not available, retrying later...{ attempt 2 } smartTemplate-util.js:688:13
SmartTemplates 13:57:27.880  [2002 ms]	 
 spellChecker not available, retrying later...{ attempt 3 } smartTemplate-util.js:688:13
SmartTemplates 13:57:29.881  [2001 ms]	 
 spellChecker not available, retrying later...{ attempt 4 } smartTemplate-util.js:688:13
SmartTemplates 13:57:31.883  [2002 ms]	 
 spellChecker not available, retrying later...{ attempt 5 } smartTemplate-util.js:688:13
Unknown localization message message_default_+title ExtensionCommon.jsm:2070
```
Restarting Thunderbird, I see this in the console:
```
SmartTemplates 13:59:29.839  [23 ms]	 
 readStringFile() - Failure: Win error 2 during operation open on file C:\Users\xxxxxxx\AppData\Roaming\Thunderbird\Profiles\xxxxxxxx.default\extensions\smartTemplates.json (The system cannot find the file specified.

) smartTemplate-util.js:688:13
```

Here is my complete Template for new messages:
```
%spellcheck(en-CA,fr)%<div style="font-family:'Trebuchet MS',sans-serif;font-size:10pt">%cursor%
</div>
```

**Comment by RealRaven2000** [comment]
So I tested this here on a test profile in Thunderbird 102.6.1 and the last lines where it works for me here:

```
SmartTemplates 14:20:12.721  [3 ms]	 
 readStringFile() - Success [smartTemplate-util.js:688:13](chrome://smarttemplate4/content/smartTemplate-util.js)
SmartTemplates 14:20:12.721  [0 ms]	 
 promise2.then populateMenus() [obsolete] [smartTemplate-util.js:688:13](chrome://smarttemplate4/content/smartTemplate-util.js)
SmartTemplates 14:20:14.650  [1929 ms]	 
 Setting spellchecker / document language to: en-CA,fr
```

It also seems to set the spellchecker correctly:
![image](https://user-images.githubusercontent.com/6314674/212342203-d8c0f15e-1ec6-4a10-a1d7-11159807952f.png)


I tested it both with Write and Reply. 

Once I disabled the global option ins Settings / Composition / "Enabled Spellcheck as you type" I do get this error message:

![image](https://user-images.githubusercontent.com/6314674/212343738-f1e5970e-ae9e-41f8-b111-929a87c8827e.png)

(the message box has no caption here)

However a user would probably expect the `%spellcheck()%` variable to enable "spellcheck as you type" for them.

Regarding the message:

```
SmartTemplates 13:59:29.839  [23 ms]	 
 readStringFile() - Failure: Win error 2 during operation open on file C:\Users\xxxxxxx\AppData\Roaming\Thunderbird\Profiles\xxxxxxxx.default\extensions\smartTemplates.json (The system cannot find the file specified.

) smartTemplate-util.js:688:13
```

- do you have any external (html) templates configured for your write / reply / forward menus? You can add them using the "configure menu items" command on the dropdown of any compose / reply / forward button:

![image](https://user-images.githubusercontent.com/6314674/212344353-4c20e013-9770-4d7d-b866-600a8eb4c386.png)

**Comment by RealRaven2000** [comment]
I have re-titled the issue and made a fix (for various issues). The strange dialog title was simply an effect of a typo in the message xhtml file. The error about BondOpenPGP.isEnabled was because of a change in Tb102, so I now do the correct processing in this case as well (during `NotifyComposeBodyReady()` when a new mail is written).

Finally I managed to force switching on "spellcheck as you type" if `%spellcheck()%` is used with any parameter other than `off`. The 

So here is a new test version:

[smartTemplate-fx-3.15.2pre10.zip](https://github.com/RealRaven2000/SmartTemplates/files/10412865/smartTemplate-fx-3.15.2pre10.zip)

To install this, as always download the zip file and then drag that file into Thunderbird Add-ons Manager (without extracting contents).

**Comment by RealRaven2000** [comment]
I also made a test with an invalid entry - `%spellcheck(en-CA,fr,xx)%` - which just creates an unspecific ecxeption. So I am also dealing with this better in this version (the missing dictionary will be listed in error console as long as at least one language can be found)

[smartTemplate-fx-3.15.2pre14.zip](https://github.com/RealRaven2000/SmartTemplates/files/10413442/smartTemplate-fx-3.15.2pre14.zip)

To install this, as always download the zip file and then drag that file into Thunderbird Add-ons Manager (without extracting contents).

**Comment by goozak** [comment]
With the pre14 version, I don't get the error anymore when "Enabled Spellcheck as you type" is disabled.  👍🏻
The last entry in the Console is weird, though, since spellcheck does indeed work...  (I have LanguageTool off, so I'm sure it's TB's spellcheck)
```
SmartTemplates [logTime init]
 SmartTemplate4.init() smartTemplate-util.js:688:13
SmartTemplates 9:35:10.122  [1 ms]	 
 SmartTemplate4.init() ends. smartTemplate-util.js:688:13
SmartTemplates 9:35:10.413  [291 ms]	 
 NotifyComposeBodyReady smartTemplate-util.js:688:13
SmartTemplates 9:35:10.628  [215 ms]	 
 SmartTemplate4.notifyComposeBodyReady()
gMsgCompose type: 0 smartTemplate-util.js:688:13
SmartTemplates 9:35:10.631  [3 ms]	 
 Enabled automatic spellcheck smartTemplate-util.js:688:13
SmartTemplates 9:35:10.631  [0 ms]	 
 spellChecker not available, retrying later...{ attempt 1 } smartTemplate-util.js:688:13
SmartTemplates 9:35:10.640  [9 ms]	 
 No deferred variables so we do not setup listeners... smartTemplate-util.js:688:13
SmartTemplates 9:35:10.772  [132 ms]	 
 SmartTemplate4.composer.load smartTemplate-util.js:688:13
SmartTemplates 9:35:10.773  [1 ms]	 
 composer.initTemplateMenu() ... smartTemplate-util.js:688:13
SmartTemplates 9:35:10.773  [0 ms]	 
 composer.initSnippetMenu() ... smartTemplate-util.js:688:13
SmartTemplates 9:35:12.635  [1862 ms]	 
 Enabled automatic spellcheck smartTemplate-util.js:688:13
SmartTemplates 9:35:12.636  [1 ms]	 
 Setting spellchecker / document language to: en-CA,fr smartTemplate-util.js:688:13
SmartTemplates 9:35:12.646  [10 ms]	 
 Disabling automatic spellcheck according to global setting. smartTemplate-util.js:688:13
```

Just to be picky 😉, an unknown language is not written to the console (although we see all the languages a few rows up):
```
SmartTemplates 9:43:0.664  [1955 ms]	 
 Enabled automatic spellcheck smartTemplate-util.js:688:13
SmartTemplates 9:43:0.665  [1 ms]	 
 Setting spellchecker / document language to: en-CA,fr,qqqq smartTemplate-util.js:688:13
SmartTemplates 9:43:0.676  [11 ms]	 
 Disabling automatic spellcheck according to global setting. smartTemplate-util.js:688:13
%spellcheck()% didn't find the following language entries: 
smartTemplate-util.js:2596:17
```
[There is nothing after "... following language entries: "]

About the missing file error (`...\smartTemplates.json (The system cannot find the file specified.)`):
> do you have any external (html) templates configured for your write / reply / forward menus?

No, I don't have any templates.

**Comment by RealRaven2000** [comment]
> Just to be picky 😉, an unknown language is not written to the console (although we see all the languages a few rows up):

You're right! I made a change but somehow it didn't go into the last build - this one works:

[smartTemplate-fx-3.15.2pre15.zip](https://github.com/RealRaven2000/SmartTemplates/files/10428730/smartTemplate-fx-3.15.2pre15.zip)

> About the missing file error (...\smartTemplates.json (The system cannot find the file specified.)):
> > do you have any external (html) templates configured for your write / reply / forward menus?
> No, I don't have any templates.

Ok then the error is nothing to worry about - the file gets created as soon as you add the first external template. It's probably not great that I log the error, but it's just a red herring in this case.

**Comment by RealRaven2000** [comment]
Implemented  in version 3.16 published on 6th of March 2023
